Crash with Spelunkery mod

### Basic questions:

- [X] I am actually using ketting, and not any other software (e.g. magma)
- [X] I am using the latest Ketting version at the time of writing for the Minecraft version, that I am using
- [X] I have provided full server logs and not a crash-report
- [X] I have checked all items without reading.
- [X] I have provided proper reproduction steps
- [X] I have ticked all boxes

### What Minecraft Version is your Server on?

1.20.1

### Full Ketting Version

1.20.1-47.3.5-0.3.3

### Description of behavior

Spelunkery mod cannot let ketting server launch.

### Reproduction Steps

```markdown
1. Use spelunkery 0.3.5 and moonlight 2.12.6
2. Run the server
3. See the mixin error
```


### Logs

[11:11:15] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher running: args [--launchTarget, forgeserver, --fml.forgeVersion, 47.3.5-0.3.3, --fml.mcVersion, 1.20.1, --fml.forgeGroup, org.kettingpowered.server, --fml.mcpVersion, 20230612.114412]
[11:11:15] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher 10.0.9+10.0.9+main.dcd20f30 starting: java version 17.0.11 by Azul Systems, Inc.; OS Windows 10 arch amd64 version 10.0
[11:11:15] [main/INFO] [ne.mi.fm.lo.ImmediateWindowHandler/]: ImmediateWindowProvider not loading because launch target is forgeserver
[11:11:15] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=union:/C:/Users/tskimiseiran/Desktop/kettingtest/libraries/org/spongepowered/mixin/0.8.5/mixin-0.8.5.jar%2388!/ Service=ModLauncher Env=SERVER
[11:11:15] [main/INFO] [ne.mi.fm.lo.mo.AbstractModProvider/]: [Ketting] Force including module MixinExtras from [].
[11:11:15] [main/INFO] [ne.mi.fm.lo.mo.AbstractModProvider/]: [Ketting] Force including module MixinExtras.beta._10 from [].
[11:11:15] [main/INFO] [ne.mi.fm.lo.mo.AbstractModProvider/]: [Ketting] Force including module MixinExtras from [].
[11:11:15] [main/INFO] [ne.mi.fm.lo.mo.JarInJarDependencyLocator/]: Found 2 dependencies adding them to mods collection
[11:11:16] [main/INFO] [mixin/]: Compatibility level set to JAVA_17
[11:11:16] [main/INFO] [cp.mo.mo.LaunchServiceHandler/MODLAUNCHER]: Launching target 'forgeserver' with arguments []
[11:11:16] [main/WARN] [mixin/]: Reference map 'spelunkery-forge-refmap.json' for spelunkery.mixins.json could not be read. If this is a development environment you can ignore this message
[11:11:17] [main/WARN] [mixin/]: Error loading class: com/mojang/blaze3d/platform/InputConstants$Key (java.lang.ClassNotFoundException: com.mojang.blaze3d.platform.InputConstants$Key)
[11:11:17] [main/INFO] [or.ke.ke.co.Ketting/]: KettingCore 1.5.2 for Minecraft 1.20.1 initializing...
[11:11:17] [main/INFO] [or.ke.ke.co.in.InjectProtect/]: Booting up InjectProtect
[11:11:17] [main/INFO] [or.ke.ke.co.Ketting/]: Loading remapper...
[11:11:18] [main/INFO] [or.ke.ke.co.Ketting/]: Loading patches...
[11:11:18] [main/INFO] [or.ke.ke.co.Ketting/]: Loaded 3 patches
[11:11:18] [main/INFO] [or.ke.ke.co.de.StacktraceDeobfuscator/]: Loaded 42290 mappings [5460 mojmap, 36830 bukkit]
[11:11:18] [main/INFO] [or.ke.ke.co.Ketting/]: Loaded JDBC driver for SQLite
[11:11:19] [main/INFO] [MixinExtras|Service/]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.3.6).
[11:11:21] [main/WARN] [mixin/]: Injection warning: LVT in net/minecraft/world/entity/vehicle/AbstractMinecart::m_8119_()V has incompatible changes at opcode 300 in callback spelunkery-common.mixins.json:AbstractMinecartMixin->@Inject::rattleMinecart(Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;IIILnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;)V.
 Expected: [I, I, I, Lnet/minecraft/core/BlockPos;, Lnet/minecraft/world/level/block/state/BlockState;]
    Found: [D, D, D, F, F]
Available: [D, D, D, F, F, I, I, I, Lnet/minecraft/core/BlockPos;, Lnet/minecraft/world/level/block/state/BlockState;, D]
--------------------------------2024-07-16 11:11:21,068 Log4j2-AsyncAppenderEventDispatcher-1-Async ERROR An exception occurred processing Appender rewrite org.apache.logging.log4j.core.appender.AppenderLoggingException: java.lang.NoClassDefFoundError: Could not initialize class net.minecraft.world.item.Items
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:165)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:134)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:125)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:89)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.appender.AsyncAppenderEventDispatcher.dispatch(AsyncAppenderEventDispatcher.java:130)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.appender.AsyncAppenderEventDispatcher.dispatchAll(AsyncAppenderEventDispatcher.java:92)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.appender.AsyncAppenderEventDispatcher.run(AsyncAppenderEventDispatcher.java:74)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.minecraft.world.item.Items
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at MC-BOOTSTRAP/kettingcore@1.5.2/org.kettingpowered.ketting.common.deobf.StacktraceDeobfuscator.deobf(StacktraceDeobfuscator.java:77)
        at MC-BOOTSTRAP/kettingcore@1.5.2/org.kettingpowered.ketting.common.deobf.StacktraceDeobfuscator.deobf(StacktraceDeobfuscator.java:53)
        at MC-BOOTSTRAP/net.minecraftforge.fmlloader@1.20.1-47.3.5-0.3.3/org.kettingpowered.ketting.deobf.DeobfRewritePolicy.rewrite(DeobfRewritePolicy.java:26)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.appender.rewrite.RewriteAppender.append(RewriteAppender.java:81)
        at MC-BOOTSTRAP/org.apache.logging.log4j.core@2.19.0/org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:161)
        ... 6 more
Caused by: java.lang.ExceptionInInitializerError: Exception org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered [in thread "main"]
        at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
        at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250)
        at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131)
        at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50)
        at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113)
        at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219)
        at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229)
        at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219)
        at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.item.Items.<clinit>(Items.java:749)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.block.ComposterBlock.m_51988_(ComposterBlock.java:66)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.Bootstrap.m_135870_(Bootstrap.java:47)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.Main.main(Main.java:163)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at MC-BOOTSTRAP/net.minecraftforge.fmlloader@1.20.1-47.3.5-0.3.3/net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111)
        at MC-BOOTSTRAP/net.minecraftforge.fmlloader@1.20.1-47.3.5-0.3.3/net.minecraftforge.fml.loading.targets.CommonLaunchHandler.serverService(CommonLaunchHandler.java:103)
        at MC-BOOTSTRAP/net.minecraftforge.fmlloader@1.20.1-47.3.5-0.3.3/net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$makeService$0(CommonServerLaunchHandler.java:27)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.run(Launcher.java:108)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.main(Launcher.java:78)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
        at cpw.mods.bootstraplauncher@1.1.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.kettingpowered.launcher.Main.main(Main.java:215)

----------------------------
A critical error has occurred and your server was shut down.
Please send this info to the Ketting team on Github
Please also include the full log file!
------------------------------------------------------------
Error type: Mixin related error
Caused by: spelunkery-common.mixins.json:AbstractMinecartMixin

Short stack trace(s):
Critical injection failure: Callback method rattleMinecart(Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;IIILnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;)V in spelunkery-common.mixins.json:AbstractMinecartMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap spelunkery-common-refmap.json
MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1362)
MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1051)

------------------------------------------------------------

### Additional context

[debug.log](https://github.com/user-attachments/files/16242761/debug.log)

[latest.log](https://github.com/user-attachments/files/16242762/latest.log)

--- a/patches/minecraft/net/minecraft/world/entity/vehicle/AbstractMinecart.java.patch
+++ b/patches/minecraft/net/minecraft/world/entity/vehicle/AbstractMinecart.java.patch
@@ -96,16 +96,25 @@
                 this.ejectPassengers();
                 if (flag && !this.hasCustomName()) {
                    this.discard();
-@@ -261,6 +_,14 @@
+@@ -260,7 +_,23 @@
+       return this.flipped ? this.getDirection().getOpposite().getClockWise() : this.getDirection().getClockWise();
     }
  
++   // Ketting start - Fix Mixin issue
++   double prevX;
++   double prevY;
++   double prevZ;
++   float prevYaw;
++   float prevPitch;
++   // Ketting end - Fix Mixin issue
++
     public void tick() {
 +      // CraftBukkit start
-+      double prevX = this.getX();
-+      double prevY = this.getY();
-+      double prevZ = this.getZ();
-+      float prevYaw = this.getYRot();
-+      float prevPitch = this.getXRot();
++      prevX = this.getX();
++      prevY = this.getY();
++      prevZ = this.getZ();
++      prevYaw = this.getYRot();
++      prevPitch = this.getXRot();
 +      // CraftBukkit end
 +
        if (this.getHurtTime() > 0) {

